### PR TITLE
Restores mapping of "stand-in" `JContractIdKey` in hollow account finalization

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
 - Restores previous behavior mapping an empty `JContractIdKey` to a `Key{contractID=0.0.0}` protobuf key.

### CODE REVIEW

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM !

**Review by iwsimon — DISMISSED** [review]
LGTM

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/legacy/core/jproto/JKey.java [hunk 1]
```diff
@@ -327,7 +327,7 @@ public static Key convertJKey(JKey jkey, int depth) throws InvalidKeyException {
                     .setThresholdKey(ThresholdKey.newBuilder().setKeys(keys).setThreshold(thd))
                     .build();
             return result;
-        } else if (jkey.hasKeyList()) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/legacy/core/jproto/JKey.java:330]: This threw NPE on line 331 without the if.
If there is no keylist, why are we getting the keylist?

File: hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/legacy/core/jproto/JKey.java [hunk 2]
```diff
@@ -312,7 +312,7 @@ public static Key convertJKey(JKey jkey, int depth) throws InvalidKeyException {
         if (depth > MAX_KEY_DEPTH) {
             throw new InvalidKeyException("Exceeding max expansion depth of " + MAX_KEY_DEPTH);
         }
-        if (!(jkey.hasThresholdKey() || jkey.hasKeyList() || jkey.isEmpty())) {
```
Comment (jsync-swirlds) [inline hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/legacy/core/jproto/JKey.java:315]: This triggered an infinite recursion without isEmpty.
Do we need special handling for empty JKeys?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/Create2OperationSuite.java [hunk 3]
```diff
@@ -22,8 +22,7 @@
 import static com.hedera.services.bdd.spec.HapiPropertySource.asSolidityAddress;
 import static com.hedera.services.bdd.spec.HapiPropertySource.contractIdFromHexedMirrorAddress;
 import static com.hedera.services.bdd.spec.HapiPropertySource.literalIdFromHexedMirrorAddress;
-import static com.hedera.services.bdd.spec.HapiSpec.defaultHapiSpec;
-import static com.hedera.services.bdd.spec.HapiSpec.propertyPreservingHapiSpec;
+import static com.hedera.services.bdd.spec.HapiSpec.*;
```
Comment (jsync-swirlds) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/opcodes/Create2OperationSuite.java:25]: Wildcard import added; probably automatically.

**Review by tinker-michaelj — COMMENTED** [review]
LGTM

**Review by jsync-swirlds — APPROVED** [review]

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**